jdaviz/app.py

```diff
@@ -1655,11 +1655,11 @@ def _update_viewer_reference_name(
             viewer_item['name'] = new_reference
 
         # optionally update the viewer IDs:
-        if update_id and viewer_item['id'] == old_reference:
-            # update the id as well
+        if update_id:
```

Comment:
Looks like you added this logic back in https://github.com/spacetelescope/jdaviz/pull/2338 . Why was the `==` check necessary back then but not anymore?

Reply:
the original implementation in lcviz had just the `==` logic without the `update_id` switch, so I think it just came with that accidentally without thinking about if we still needed it.

Reply:
It wasn't necessary then or now, hence the bugfix.